crmsh/xmlutil.py

```diff
@@ -243,28 +303,33 @@ def rsc_clone(self, ident):
     def is_managed(self, ident):
         '''
         Is this resource managed?
+        Return (boolean, reason)
         '''
         rsc_node = self.rsc2node(ident)
-        if rsc_node is None:
-            return False
         # maintenance-mode, if true, overrides all
         attr = get_attr_value(self.prop_elem, "maintenance-mode")
         if attr and is_xs_boolean_true(attr):
-            return False
-        # then check the rsc is-managed meta attribute
+            return False, "cluster property maintenance-mode is true"
+        # then check if all nodes are in maintenance
+        if NodeState().are_all_nodes_in_maintenance():
+            return False, "all nodes are in maintenance"
+        # then check if node running this resource is in maintenance
+        if NodeState().is_node_running_resource_in_maintenance(ident):
+            return False, "node which running \"{}\" is in maintenance".format(ident)
         rsc_meta_node = get_rsc_meta_node(rsc_node)
+        # then check the rsc maintenance meta attribute
+        attr = get_attr_value(rsc_meta_node, "maintenance")
+        if attr and is_xs_boolean_true(attr):
+            return False, "resource \"{}\" is in maintenance".format(ident)
```

Comment:
The above "maintenance" checks are not the scope of is_managed(). Since, "maintenance" and "is-managed" are two different terminologies and not fit into the current function name itself all together. I'm thinking their function names something like,

is_managed(rsc)
is_maintenance(rsc)
is_managed_or_maintenance(rsc)

Reply:
From display of `crm_mon`, pacemaker show both situations as `unmanaged` for the resource
How about change the function name as `is_managed_or_maintenance`?